lang/cs/Org.Apache.REEF.Client/AzureBatch/Service/AzureBatchService.cs

```diff
@@ -39,21 +41,40 @@ public sealed class AzureBatchService : IDisposable
         public string PoolId { get; private set; }
 
         private BatchClient Client { get; set; }
+        private ContainerRegistryProvider ContainerRegistryProvider { get; set; }
+        private IList<string> Ports { get; set; }
+        private ICommandBuilder CommandBuilder { get; set; }
+        private bool AreContainersEnabled
+        {
+            get
+            {
+                return this.ContainerRegistryProvider.IsValid();
+            }
+        }
+
         private bool disposed;
 
         [Inject]
         public AzureBatchService(
+            ITcpPortProvider portProvider,
+            ContainerRegistryProvider containerRegistryProvider,
```

Comment:
remove ITcpPortProvider in constructor?

Reply:
Right! will remove it.

---
In reply to: [201768491](https://github.com/sharathmalladi/reef/pull/5#discussion_r201768491) [](ancestors = 201768491)